### DISCUSSION TITLE
Where does a package's source/config etc come from?

### DISCUSSION THREAD

**Discussion by GeoffreyPlitt** [Homebrew/discussions#2292] [Writing Formulae/Casks] [answered]
One of my favorite packages was updated in github, but the Homebrew target is not installing it right. The package maintainers seem to busy to update the Homebrew target, but I have some time to do it.

Where are the Homebrew package configurations stored, so that I can update it?

**Comment by tlk** [comment, marked as answer]
You may be looking for [how to update/maintain Homebrew formulas](https://docs.brew.sh/Formula-Cookbook).

The command line tool "ack", for example, has a Homebrew formula that can be found on Github here: https://github.com/Homebrew/homebrew-core/blob/master/Formula/ack.rb and on the file system with this command: `file $(brew --prefix)/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/ack.rb`

☙

For some software Homebrew uses another concept called "Homebrew casks".

The Firefox package, for example, can be found on Github here: https://github.com/Homebrew/homebrew-cask/blob/master/Casks/firefox.rb and the same "cask" file can be located on a system with Homebrew installed by running
`file $(brew --prefix)/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/firefox.rb`

See https://github.com/Homebrew/homebrew-cask/blob/master/CONTRIBUTING.md#updating-a-cask

----

(I'm not a maintainer, I have just been following the project for a bit)

**Reply by GeoffreyPlitt** [reply]
thank you!
